Add Top SQL doc #8123 (#7418)
pingcap/docs: dashboard/top-sql.md

---
title: Top SQL
summary: This document describes how to use Top SQL to locate SQL queries that contribute to a high load.
---

# Top SQL

> **Warning:**
>

> > Currently, Top SQL is an experimental feature. It is not recommended that you use it for production environments.

Reviewer: 另一篇文档我都改成了 XX is an experimental feature in v5.4.0 and is not recommended for use in production. 看看要不要这么用。

Author: 这是参考了目前英文文档里大部分实验特性的写法，所以感觉可以不改。

> This document describes how to use Top SQL to locate SQL queries that contribute to a high load of a TiDB or TiKV node in a specified time range. For example, you can use Top SQL to locate an analytic query that contributes to 99% of the load for a low-load database.

Reviewer 2: ```suggestion
This document describes how to use Top SQL to locate SQL queries that contribute to a high load of a TiDB or TiKV node in a specified time range. For example, you can use Top SQL to locate an analytic query that consumes to 99% of the load for a low-load database.
```

> For a specified TiDB or TiKV node, Top SQL can provide the following features:

Reviewer: ```suggestion
For a specified TiDB or TiKV node, Top SQL provides the following features:
```

* Show the top 5 types of SQL queries that contribute the most to the load in a specified time range.

> * Show information such as CPU usage, requests per second, average latency, and query plan of a particular query, which can be used as a potential performance optimization point to improve your business.

Reviewer: ```suggestion
* Show information such as CPU usage, requests per second, average latency, and query plan of a particular query, which can be used for potential performance optimization to improve your business.
```

## Enable Top SQL

The Top SQL feature is disabled by default. You can enable the feature for the entire cluster using either of the following methods:

- Method 1: Log in to TiDB Dashboard, click **Top SQL** in the left pane, click the gear button in the upper-right corner of the page, and then enable the Top SQL feature.
- Method 2: Set the value of the TiDB system variable [`tidb_enable_top_sql`](/system-variables.md#tidb_enable_top_sql-introduced-from-v540-version) to `ON`.

> **Note:**
>
> Enabling Top SQL has a slight impact on cluster performance.

> ## Using Top SQL

Reviewer: ```suggestion
## Use Top SQL
```

Once Top SQL is enabled, you can log into TiDB Dashboard, and then click **Top SQL** in the left pane to use it.

![Top SQL](/media/dashboard/top-sql-overview.png)

Usage tips：

> * You can select the target node and time range in the drop-down list at the top of the page, or you can select the time range in the chart.

Reviewer 2: ```suggestion
* You can select the target node and time range in the drop-down menu at the top of the page, or you can select the time range in the chart.
```

Author: ```suggestion
* You can select the target node and time range in the drop-down lists at the top of the page, or you can select the time range in the chart.
```

Author: @SunRunAway The list of selectable items with an arrow button is usually called a drop-down list in technical writing. See https://en.wikipedia.org/wiki/Drop-down_list.

> * You can select the target node and time range in the drop-down lists at the top of the page, or you can select the time range in the chart.

Reviewer 2: ```suggestion
* You can select the target node and time range in the drop-down menu at the top of the page, or you can select the time range in the chart.
```

We usually call it "drop-down menu"

Author: Do you have a refererence? I searched "drop-down menu" in our TiDB English docs and got no results.

Reviewer 2: https://getbootstrap.com/docs/4.0/components/dropdowns/

Author: Thanks. I see "dropdown-menu" is used at the code level for R&D but we are using "drop-down list" in user documents usually for users. See the IBM Style Guide as follows:

![image](https://user-images.githubusercontent.com/79440533/150900927-3f0c2f71-e243-4da3-9543-a50e480a88f5.png)

> * If the data in the chart is out of date, you can click **Refresh**, or select auto-refresh and specify the auto-refresh interval in the **Refresh** drop-down list.

Reviewer 2: ```suggestion
* If the data in the chart is out of date, you can click **Refresh** button, or select auto-refresh and specify the auto-refresh interval in the **Refresh** drop-down menu.
```

Author: @SunRunAway 
**Refresh** does not look like a button because it has a drop-down arrow on the right, how about I change this paragraph as follows?

```suggestion
* If the data in the chart is out of date, you can click **Refresh** at the top of the page, or click the drop-down arrow beside **Refresh** to specify an auto-refresh interval.
```

Reviewer 2: The **Refresh**  is conbined with a button and a drop-down menu
So when the users want to refresh, they can click the **Refresh** button, or they can chooss the auto-refresh interval in the **Refresh** drop-down menu.

Author: If we mention both **Refresh** button and **Refresh** drop-down menu in documents, it might give users an impression that there are two different components in different locations, so I think it is better to keep the original content.

> * If the data in the chart is out of date, you can click **Refresh**, or you can select whether to refresh automatically and select the auto-refresh interval in the **Refresh** drop-down list.

Reviewer: ```suggestion
* If the data in the chart is out of date, you can click **Refresh**, or select auto-refresh and specify the auto-refresh interval in the **Refresh** drop-down list.
```

* The chart shows the top 5 types of queries that contribute the most to the load of the selected node in the selected time range.

> * You can select a query type in the list to view query details, including the execution plan for that query, the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call for the plan on this node.

Author: ```suggestion
* You can select a query type in the list to view query details, including the execution plan for that query, the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call of the execution plan on this node.
```

> * You can click a query type in the list to view query details, including the execution plan for that query, and the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call for each execution plan on this node.

Reviewer 2: ```suggestion
* You can select a query type in the list to view query details, including the execution plan for that query, and the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call for each execution plan on this node.
```

Reviewer 2: 这句有两个and..

> * You can select a query type in the list to view query details, including the execution plan for that query, and the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call for each execution plan on this node.

Author: ```suggestion
* You can select a query type in the list to view query details, including the execution plan for that query, the Call/sec, Scan Rows/sec, Scan Indexes/sec, and Latency/call for each execution plan on this node.
```
